### ISSUE TITLE
Website for sfnetworks documentation

### ISSUE THREAD

**Issue by Robinlovelace** [luukvdmeer/sfnetworks#2]
- [x] Get continuous integration service working, I suggest GH actions
- [x] Create gh-pages branch and add website created by pkgdown
- [x] Make website update with every commit to master

**Comment by Robinlovelace** [comment]
Good link on setting-up CI + automated documentation via travis: https://pkgdown.r-lib.org/reference/deploy_site_github.html happy to help with this.

**Comment by Robinlovelace** [comment]
I'm up for looking at this now. For it to work @luukvdmeer will need to have a repo called luukvdmeer.github.io with a file called `index.html` for the website to be published by gh-pages.

**Comment by Robinlovelace** [comment]
The first checkbox can be ticked when this is merged, pending tests passing: https://github.com/luukvdmeer/sfnetworks/pull/34

**Comment by Robinlovelace** [comment]
See here for example of deploy script. I plan to give that a go after #34 is merged.

**Comment by luukvdmeer** [comment]
The pkgdown deployment was a much bigger hassle than I hoped for. The majority of packages still seems to use Travis, so it was hard to find good examples with GH actions.
 
The `vctrs` example @Robinlovelace mentioned above turned out to be too simple for spatial packages.

Some lessons learned:

1.
The `pkgdown.yaml` file also needs the [section](https://github.com/luukvdmeer/sfnetworks/blob/66b107f9a732d45df549c21d591c6544dff260f8/.github/workflows/pkgdown.yaml#L23) where all system dependencies, like gdal, are installed.

2.
On macOS there can occur problems with the `xml2` dependency of `pkgdown`. Installing that package first before installing other dependencies helped me out:
```r
remotes::install_github("r-lib/xml2")
```
 See [here](https://github.com/r-lib/xml2/issues/223) for some background. Seems like a mess!

3.
Using the most recent version of `pkgdown` failed during the Deploy Package phase with:
```
fatal: could not read Username for 'https://github.com': Device not configured
```
What solved it for me was using an older pkgdown version, just like in the `vctrs` example:
```r
remotes::install_github("jimhester/pkgdown@github-actions-deploy")
```
Instead of:
```r
remotes::install_dev("pkgdown")`.
```

4.
However, this did not solve all Git issues. A new issue occured: 
```remote: Invalid username or password. fatal: Authentication failed for 'https://github.com/luukvdmeer/sfnetworks.git/'
``` 
This seems to relate to this [pkgdown issue](https://github.com/r-lib/pkgdown/issues/1275), which is solved for Travis, but for me still occurred with GH actions. To solve this, I again used the way `vctrs` does this, with `deploy_local()` in combination with a GitHub Personal Access Token (PAT) and GitHub Secrets: 
```r
{ url <- 'https://x-access-token:${{secrets.DEPLOY_PAT}}@github.com/${{github.repository}}.git'; pkgdown:::deploy_local(new_process = FALSE, remote_url = url) }
```
Instead of:
```r
pkgdown::deploy_to_branch(new_process = FALSE)`
```
The latter does require some steps that should be done beforehand: creating a custom GitHub Personal Access Token and with this token creating a 'secret' for the GitHub repository of the package. See [here](https://github.com/maxheld83/ghpages#secrets) for a short explanation (I used `DEPLOY_PAT` as secret name instead of `GH_PAT`, just like in the `vctrs` example).

I am not sure if I followed the best practices, and since GH actions is still very new, a lot will probably change soon, but at least it worked for now with the way I did it.

See here the resulting [pkgdown.yaml file](https://github.com/luukvdmeer/sfnetworks/blob/master/.github/workflows/pkgdown.yaml).

See #43 for an overview of how we might fine-tune the CI with GH actions in the coming weeks. For now, I am just happy it works!

**Comment by Robinlovelace** [comment]
Sounds like a hassle. FYI I recently set-up GH Actions -> deploy website via `use_tic()` form the latest dev version of the `tic` package and I think it's working on this package with sf deps OK: https://github.com/ITSLeeds/od

**Comment by luukvdmeer** [comment]
Seems to work good now. For fine tuning the pkgdown website, see #44
